### PR TITLE
Fix annotations for voltron tls route configuration (too long and invalid characters)

### PR DESCRIPTION
**Pull request by Brian-McM**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by peterkellydev — APPROVED** [review]
LGTM

**Review by tmjd — COMMENTED** [review]
This doesn't put any cap on the length, should there be some cap on the length. Perhaps truncating the name, you could have a slice with all the names and then check the list and verify the truncated name doesn't clash with another name being used.

File: pkg/render/manager/manager_route_config.go [hunk 1]
```diff
@@ -302,6 +305,54 @@ func (builder *voltronRouteConfigBuilder) mountConfigMapReference(name, key stri
...
+		}
+
+		// Just in case there's a collision
+		for i := 1; ; i++ {
+			if _, ok := annotations[formattedKey]; !ok {
+				break
+			}
+			suffix := strconv.Itoa(i)
+			formattedKey = formattedKey[0:len(formattedKey)-len(suffix)] + suffix
+			l := len(formattedKey)
```
Comment (tmjd) [inline pkg/render/manager/manager_route_config.go:338]: This looks like maybe something for debugging. Is this needed (and the line below)?

Comment (Brian-McM) [reply pkg/render/manager/manager_route_config.go:338]: Oh shoot, yes that wasn't supposed to be committed.

Comment (Brian-McM) [reply pkg/render/manager/manager_route_config.go]: Just removed it.

**Review by tmjd — APPROVED** [review]
LGTM

**Pull request merged**